# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1303264376 14400
#      Tue Apr 19 21:52:56 2011 -0400
# Branch stable
# Node ID 83606de30dae14cef5166ba8b6e5800ce029d04b
# Parent  6f3f189578514a0d3228b5c05e50d9c4e4e2f6d7
maint: generate ChangeLog automatically

* Move old hand-edited ChangeLog files to OLD-ChangeLogs directory.
* changelog.tmpl: New file.
* Makefile.am (BUILT_DISTFILES): Add ChangeLog to the list.
(EXTRA_DIST): Add OLD-ChangeLogs files.
(ChangeLog): New target.
* autogen.sh: Don't require ChangeLog to exist.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -21,30 +21,37 @@
 include common.mk
 
 ## Avoid making multiple subdirs in parallel which can lead 
 ## to a confusing error message stream
 .NOTPARALLEL:
 
 ACLOCAL_AMFLAGS = -I m4
 
-BUILT_DISTFILES = AUTHORS BUGS INSTALL.OCTAVE
+BUILT_DISTFILES = AUTHORS BUGS ChangeLog INSTALL.OCTAVE
 
 EXTRA_DIST = \
   AUTHORS \
   BUGS \
   COPYING \
   ChangeLog \
-  ChangeLog.1 \
   INSTALL \
   INSTALL.OCTAVE \
   NEWS \
   NEWS.1 \
   NEWS.2 \
   NEWS.3 \
+  OLD-ChangeLogs/ChangeLog \
+  OLD-ChangeLogs/ChangeLog.1 \
+  OLD-ChangeLogs/doc-ChangeLog \
+  OLD-ChangeLogs/libcruft-ChangeLog \
+  OLD-ChangeLogs/liboctave-ChangeLog \
+  OLD-ChangeLogs/scripts-ChangeLog \
+  OLD-ChangeLogs/src-ChangeLog \
+  OLD-ChangeLogs/test-ChangeLog \
   PROJECTS \
   README \
   README.Cygwin \
   README.Linux \
   README.MacOS \
   README.Windows \
   README.kpathsea \
   autogen.sh \
@@ -151,16 +158,21 @@ run-octave: run-octave.in Makefile
 	else \
 	  cp $< $@; \
 	fi
 
 AUTHORS BUGS INSTALL.OCTAVE:
 	$(MAKE) -C doc/interpreter ../../$@
 .PHONY: AUTHORS BUGS INSTALL.OCTAVE
 
+ChangeLog:
+	(cd $(srcdir); hg log --style=changelog.tmpl --prune=b0e60ad4ae26ec2ca3567a17b29a53e0cd2051d8 --branch=`hg branch`; echo ""; echo "See the files in the OLD-ChangeLogs directory for older changes") > $@.t
+	mv $@.t $@
+.PHONY: ChangeLog
+
 octetc_DATA = NEWS
 
 DIRS_TO_MAKE = \
   $(localfcnfiledir) \
   $(localapifcnfiledir) \
   $(localverfcnfiledir) \
   $(localoctfiledir) \
   $(localapioctfiledir) \
diff --git a/ChangeLog b/OLD-ChangeLogs/ChangeLog
rename from ChangeLog
rename to OLD-ChangeLogs/ChangeLog
diff --git a/ChangeLog.1 b/OLD-ChangeLogs/ChangeLog.1
rename from ChangeLog.1
rename to OLD-ChangeLogs/ChangeLog.1
diff --git a/doc/ChangeLog b/OLD-ChangeLogs/doc-ChangeLog
rename from doc/ChangeLog
rename to OLD-ChangeLogs/doc-ChangeLog
diff --git a/libcruft/ChangeLog b/OLD-ChangeLogs/libcruft-ChangeLog
rename from libcruft/ChangeLog
rename to OLD-ChangeLogs/libcruft-ChangeLog
diff --git a/liboctave/ChangeLog b/OLD-ChangeLogs/liboctave-ChangeLog
rename from liboctave/ChangeLog
rename to OLD-ChangeLogs/liboctave-ChangeLog
diff --git a/scripts/ChangeLog b/OLD-ChangeLogs/scripts-ChangeLog
rename from scripts/ChangeLog
rename to OLD-ChangeLogs/scripts-ChangeLog
diff --git a/src/ChangeLog b/OLD-ChangeLogs/src-ChangeLog
rename from src/ChangeLog
rename to OLD-ChangeLogs/src-ChangeLog
diff --git a/test/ChangeLog b/OLD-ChangeLogs/test-ChangeLog
rename from test/ChangeLog
rename to OLD-ChangeLogs/test-ChangeLog
diff --git a/autogen.sh b/autogen.sh
--- a/autogen.sh
+++ b/autogen.sh
@@ -10,17 +10,17 @@ set -e
 AUTOMAKE="automake --foreign --warnings=no-portability"
 export AUTOMAKE
 
 ## Check for files that automake --gnu would normally look for, except
 ## AUTHORS, which we autogenerate from the documentation files along with
 ## building the rest of Octave, and INSTALL, which is linked from
 ## gnulib/doc/INSTALL by the bootstrap script.
 
-for f in NEWS README ChangeLog COPYING; do
+for f in NEWS README COPYING; do
   if ! test -f $f; then
     echo "required file $f is missing" 2>&1
     exit 1
   fi
 done
 
 echo "generating source lists for liboctave/Makefile..."
 
diff --git a/changelog.tmpl b/changelog.tmpl
new file mode 100644
--- /dev/null
+++ b/changelog.tmpl
@@ -0,0 +1,2 @@
+header = '{date|shortdate}  {author|person}  <{author|email}>\n\n'
+changeset = '\t{desc|tabindent|strip}\n\n\tFiles: {files|stringify|fill68|tabindent|strip}\n\n'
